test/refactoring/ConvertAsync/convert_pattern.swift

```diff
@@ -0,0 +1,411 @@
+  }
+
+  // OUT-OF-LINE:      let strs = try await stringTupleParam()
+  // OUT-OF-LINE-NEXT: let (str1, str2) = strs
+  // OUT-OF-LINE-NEXT: print(str1, str2, strs)
+
+  // RUN: %refactor-check-compiles -convert-call-to-async-alternative -dump-text -source-filename %s -pos=%(line+1):3 | %FileCheck -check-prefix=OUT-OF-LINE-VAR %s
+  stringTupleParam { strs, err in
+    guard var (str1, _) = strs else { return }
+    str1 = ""
+    print(str1, {strs!})
+  }
+
+  // OUT-OF-LINE-VAR:      let strs = try await stringTupleParam()
+  // OUT-OF-LINE-VAR-NEXT: var (str1, _) = strs
+  // OUT-OF-LINE-VAR-NEXT: str1 = ""
+  // OUT-OF-LINE-VAR-NEXT: print(str1, {strs})
+
+  // RUN: %refactor-check-compiles -convert-call-to-async-alternative -dump-text -source-filename %s -pos=%(line+1):3 | %FileCheck -check-prefix=OUT-OF-LINE-CASE %s
+  stringTupleParam { strs, err in
+    guard case (let str1, var str2)? = strs else { return }
+    print(str1, str2, strs!)
+  }
+
+  // OUT-OF-LINE-CASE:      let strs = try await stringTupleParam()
+  // OUT-OF-LINE-CASE-NEXT: var (str1, str2) = strs
+  // OUT-OF-LINE-CASE-NEXT: print(str1, str2, strs)
+
+  // RUN: %refactor-check-compiles -convert-call-to-async-alternative -dump-text -source-filename %s -pos=%(line+1):3 | %FileCheck -check-prefix=FALLBACK %s
+  stringTupleParam { strs, err in
+    guard let (str1, str2) = strs else { return }
+    print(str1, str2)
+    if .random(), err == nil {
+      print("yay")
+    } else {
+      print("nay")
+    }
+  }
+
+  // FALLBACK:      var strs: (String, String)? = nil
+  // FALLBACK-NEXT: var err: Error? = nil
+  // FALLBACK-NEXT: do {
+  // FALLBACK-NEXT:   strs = try await stringTupleParam()
+  // FALLBACK-NEXT: } catch {
+  // FALLBACK-NEXT:   err = error
+  // FALLBACK-NEXT: }
+  // FALLBACK-EMPTY:
+  // FALLBACK-NEXT: guard let (str1, str2) = strs else { <#return#> }
+  // FALLBACK-NEXT: print(str1, str2)
+  // FALLBACK-NEXT: if .random(), err == nil {
+  // FALLBACK-NEXT:   print("yay")
+  // FALLBACK-NEXT: } else {
+  // FALLBACK-NEXT:   print("nay")
+  // FALLBACK-NEXT: }
+
+  // RUN: %refactor-check-compiles -convert-call-to-async-alternative -dump-text -source-filename %s -pos=%(line+1):3 | %FileCheck -check-prefix=STRING-TUPLE-RESULT %s
+  stringTupleResult { res in
+    switch res {
+    case .success((let x, let y)):
+      print(x, y)
+    case .failure:
+      print("oh no")
+    }
+  }
+  // STRING-TUPLE-RESULT:      do {
+  // STRING-TUPLE-RESULT-NEXT:   let (x, y) = try await stringTupleResult()
+  // STRING-TUPLE-RESULT-NEXT:   print(x, y)
+  // STRING-TUPLE-RESULT-NEXT: } catch {
+  // STRING-TUPLE-RESULT-NEXT:   print("oh no")
+  // STRING-TUPLE-RESULT-NEXT: }
+
+  // RUN: %refactor-check-compiles -convert-call-to-async-alternative -dump-text -source-filename %s -pos=%(line+1):3 | %FileCheck -check-prefix=STRING-TUPLE-TYPED-RESULT %s
+  stringTupleResult { res in
+    switch res {
+    case .success((let x, let y): (String, String)):
```

Comment:
I hadn't seen this before so tried it myself - I get a compile error for this and a few others in this file. I'm a little confused as to how this test isn't failing since the *other* %refactor-check-compiles should fail to compile.

Reply:
Looks like the file being compiled is... empty. I'm guessing that happened in https://github.com/apple/swift/pull/37528?

Reply:
Yep, fixed here: https://github.com/apple/swift/pull/37755